mpf/core/service_controller.py

```diff
@@ -75,4 +75,5 @@ def get_coil_map(self) -> List[CoilMap]:
 
         # sort by board + driver number
         coil_map.sort(key=lambda x: x[0] + str(x[1].hw_driver.number))
+        coil_map.sort(key=lambda x: len(x[0] + str(x[1].hw_driver.number)))
```

Comment:
I guess you want to sort by len + the string itself, right? Currently the order of 000 and 111 would be random because both have a length of 3.

Reply:
Correct me if I am wrong but I wanted to sort by board/driver number(first) AND by length(second). This can be done (one after another) because python's sort is stable. As far as I can tell (also based on the tests) the order of 000 and 111 would not be random.